### DISCUSSION TITLE
Introduce Gemfile, ruby-version and nvmrc to template

### DISCUSSION THREAD

**Discussion by barbieri** [react-native-community/discussions-and-proposals#411] [Ideas]
The `template/ios` uses `Podfile` and that is based on Ruby and CocoaPods.

To aid the process we could provide a `Gemfile` and associated `Gemfile.lock` with the recommended ruby and CocoaPods version, this way we avoid issues such as using the incorrect versions. Then we change every reference to _Execute `pod install`_ to be _Execute `bundle exec pod install`_ to make sure the correct versions are used.

By providing a `.ruby-version` file, we could help http://rbenv.org/ and https://rvm.io/ uses to use the correct version and avoid pitfalls (such as using the default macOS version, that is a universal binary and may cause issues with ffi on arm64/m1).

Along these lines, we could also provide `.nvmrc` with the tested/recommended version and even lock `package.json` with `engines: { node: "..." }`, this way we minimize problems.

I also reported it to https://github.com/react-native-community/react-native-template-typescript/issues/225, but it was said that this change should go in React-Native itself and then they will adapt in future releases.

**Comment by kelset** [comment]
This sounds reasonable - would you mind making directly a PR so that it can be discussed "code in hands"?

**Reply by barbieri** [reply]
will do!

**Reply by barbieri** [reply]
First attempt at https://github.com/facebook/react-native/pull/32303

**Comment by Adnan-Bacic** [comment]
i like the idea of having the version of Node in the project too with an .nvmrc file. is this planned to be added eventually as well?

**Comment by xiaoxiangmoe** [comment]
@barbieri Should we move .ruby-version and Gemfile to ios directory？

**Reply by barbieri** [reply]
while one could move, it's usually better hosted in the parent so everything that may use ruby in the repo uses the same base (not sure if there are any... but still)
